bitswap/src/behaviour.rs

```diff
@@ -231,7 +231,7 @@ impl NetworkBehaviour for Bitswap {
 
     fn inject_disconnected(&mut self, peer_id: &PeerId) {
         debug!("bitswap: inject_disconnected {:?}", peer_id);
-        //self.connected_peers.remove(peer_id);
+        self.connected_peers.remove(peer_id);
```

Comment:
Could add a comment explaining why stats is not removed as well? I can't guess it from the context.

Reply:
> I wonder if there are previous block reads which were requested but should now be cancelled for this peerid?

Which ones do you have in mind? The related `Ledger` is dropped, and the `get_block`-related futures are peer-agnostic.

Reply:
Oki, because of this line I was thinking that the block reads are in fact per peer: https://github.com/rs-ipfs/rust-ipfs/blob/master/src/p2p/behaviour.rs#L272

Great if those are shared already, but then I would expect there to be some kind of "ticketing" system, perhaps a `HashMap<Cid, Vec<PeerId>>`  kind of thing which would allow you to correlate back the loaded block in case it's wanted by multiple peers.

Reply:
As discussed, let's defer that for now.